FlagsmithClient/src/main/java/com/flagsmith/internal/FlagsmithApi.kt

```diff
@@ -29,52 +29,49 @@ sealed class FlagsmithApi: FuelRouting {
             }
         }
     override val body: String?
-        get() {
-            return when(this) {
-                is setTrait -> Gson().toJson(TraitWithIdentity(key = trait.key, value = trait.value, identity = Identity(identity)))
-                is postAnalytics -> Gson().toJson(eventMap)
-                else -> null
-            }
+        get() = when (this) {
+            is SetTrait -> Gson().toJson(
+                TraitWithIdentity(
+                    key = trait.key,
+                    value = trait.value,
+                    identity = Identity(identity)
+                )
+            )
+            is PostAnalytics -> Gson().toJson(eventMap)
+            else -> null
         }
 
-    override val bytes: ByteArray?
-        get() = null
+    override val bytes: ByteArray? = null
 
     override val headers: Map<String, HeaderValues>?
-        get() {
-            val headers = mutableMapOf<String, HeaderValues>("X-Environment-Key" to listOf(environmentKey ?: ""))
+        get() = mutableMapOf<String, HeaderValues>(
+            "X-Environment-Key" to listOf(environmentKey ?: "")
+        ).apply {
             if (method == Method.POST) {
-                headers["Content-Type"] = listOf("application/json")
+                this += "Content-Type" to listOf("application/json")
             }
-            return headers
         }
 
     override val method: Method
-        get() {
-            return when(this) {
-                is getIdentityFlagsAndTraits -> Method.GET
-                is getFlags -> Method.GET
-                is setTrait -> Method.POST
-                is postAnalytics -> Method.POST
-            }
+        get() = when (this) {
+            is GetIdentityFlagsAndTraits -> Method.GET
+            is GetFlags -> Method.GET
+            is SetTrait -> Method.POST
+            is PostAnalytics -> Method.POST
         }
 
     override val params: Parameters?
-        get() {
-            return when(this) {
-                is getIdentityFlagsAndTraits -> listOf("identifier" to this.identity)
-                is setTrait -> listOf("identifier" to this.identity)
-                else -> null
-            }
+        get() = when (this) {
+            is GetIdentityFlagsAndTraits -> listOf("identifier" to this.identity)
+            is SetTrait -> listOf("identifier" to this.identity)
```

Comment:
Appreciate that this was already there in the previous code but we don't actually need the identifier parameter on this request.

Reply:
Sure, I've got one more "general improvements" PR to make so I might as well fix it in that one.

Reply:
Actually as I need to make another change to this PR, I'll do it here as well.